examples/nidaqmx_analog_input/measurement.py

```diff
@@ -30,31 +44,93 @@ def measure(physical_channel, sample_rate, number_of_samples):
         sample_rate,
         number_of_samples,
     )
-
-    def cancel_callback():
-        logging.info("Canceling measurement")
-        task_to_abort = task
-        if task_to_abort is not None:
-            task_to_abort.control(nidaqmx.constants.TaskMode.TASK_ABORT)
-
-    measurement_service.context.add_cancel_callback(cancel_callback)
-    time_remaining = measurement_service.context.time_remaining
-
-    timeout = min(time_remaining, 10.0)
-    with nidaqmx.Task() as task:
-        task.ai_channels.add_ai_voltage_chan(physical_channel)
+    session_management_client = nims.session_management.Client(
+        grpc_channel=measurement_service.get_channel(
+            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
+            service_class=nims.session_management.GRPC_SERVICE_CLASS,
+        )
+    )
+    with contextlib.ExitStack() as stack:
+        reservation = stack.enter_context(
+            session_management_client.reserve_sessions(
+                context=measurement_service.context.pin_map_context,
+                pin_or_relay_names=[physical_channel],
+                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DAQMX,
+                # If another measurement is using the session, wait for it to complete.
+                # Specify a timeout to aid in debugging missed unreserve calls.
+                # Long measurements may require a longer timeout.
+                timeout=60,
+            )
+        )
+        if len(reservation.session_info) != 1:
+            measurement_service.context.abort(
+                grpc.StatusCode.INVALID_ARGUMENT,
+                f"Unsupported number of sessions: {len(reservation.session_info)}",
+            )
+        session_info = reservation.session_info[0]
+        task = stack.enter_context(_create_nidaqmx_task(session_info))
+        channel_mappings = session_info.channel_mappings
+        pending_cancellation = False
+
+        def cancel_callback():
+            logging.info("Canceling measurement")
+            session_to_abort = task
+            task_to_abort = task
+            if task_to_abort is not None:
+                nonlocal pending_cancellation
+                pending_cancellation = True
+                session_to_abort.abort()
+
+        measurement_service.context.add_cancel_callback(cancel_callback)
+        time_remaining = measurement_service.context.time_remaining
+
+        timeout = min(time_remaining, 10.0)
+        channel = next(
+            channel_mapping.channel
+            for channel_mapping in channel_mappings
+            if channel_mapping.pin_or_relay_name == physical_channel
+        )
+        task.ai_channels.add_ai_voltage_chan(channel, min_val=0, max_val=5)
```

Comment:
Ok. But I'm getting error when running the teststand sequence with `--no-use-grpc-device` option (running with `LibraryInterpreter`).
```
nidaqmx.errors.DaqError: Specified operation cannot be performed when there are no devices in the task.
Task Name: Dev1

Status Code: -200477
```
The reason for the error is,
- A task has been created already in the sequence setup `create_nidaqmx_tasks()` with grpc_options in **teststand_fixture.py**. 

- Since the measurement is run with `--no-use-grpc-device`, the created task doesn't add `ai_channel` and also the addition of the check `if not session_info.session_exists:` prevents the created task from adding `ai_channel`, which results in the error. 

However, I was able to run the teststand sequence by skipping `create_nidaqmx_tasks()` and `destroy_nidaqmx_tasks()` in sequence setup and cleanup. 


So, is skipping `create_nidaqmx_tasks()` and `destroy_nidaqmx_tasks()` in sequence setup and cleanup the expected workflow?

Reply:
The TestStand sequence requires gRPC.